packages/framework/presence/src/systemWorkspace.ts

```diff
@@ -254,8 +256,10 @@ class SystemWorkspaceImpl implements PresenceStatesInternal, SystemWorkspace {
 			// The given association is newer than the one we have.
 			// Update the order and current connection ID.
 			attendee.order = order;
+			if (attendee.getConnectionStatus() === SessionClientStatus.Disconnected) {
+				isNew = true;
+			}
 			attendee.setConnectionId(clientConnectionId);
-			isNew = true;
 		}
```

Comment:
- add comments
- maybe `isNew` could use a rename like `isNewConnection`.
- an overall comment that all updated attendees will have their connection status set to Connected seems worthy.... as it is up to the caller to reset if needed.

Reply:
I think `isJoining` (the session) is the best name here, since it dictates if we announce attendeeJoined event. If a known attendee is connected and we receive a new connectionId, we should set the new connectionId but they are not necessarily joining the session and should not be announced. If known attendee is disconnected then it makes sense to announce new connection, since they are (re)joining the session. 

Removed the connection status update here as this is handled in processUpdate

Reply:
If we don't set connection status here where should `setConnectionId` be called to silently update the status?
(Silence is bad. With something as tricky as the connection status seems to be anything that updates the status should be very clear and explicit. Default param is worst. Having it be a boolean is secondarily unfortunate.)

Reply:
See comment https://github.com/microsoft/FluidFramework/pull/23113#discussion_r1845903563